android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUIPlainSpinnerPicker.java

```diff
@@ -0,0 +1,197 @@
+		TiCompositeLayout layout = (TiCompositeLayout) getNativeView();
+		if ((pickerProxy == null) || (layout == null)) {
+			return;
+		}
+
+		// Fire a "change" event.
+		for (int index = 0; index < layout.getChildCount(); index++) {
+			if (layout.getChildAt(index) == picker) {
+				pickerProxy.fireSelectionChange(index, newValue);
+				break;
+			}
+		}
+	}
+
+	private NumberPicker getNumberPickerForIndex(int columnIndex)
+	{
+		// Fetch the view group hosting all picker column child views.
+		TiCompositeLayout layout = (TiCompositeLayout) getNativeView();
+		if (layout == null) {
+			return null;
+		}
+
+		// Validate given column index.
+		if ((columnIndex < 0) || (columnIndex >= layout.getChildCount())) {
+			return null;
+		}
+
+		// Return the column's picker view.
+		View childView = layout.getChildAt(columnIndex);
+		if (childView instanceof NumberPicker) {
+			return (NumberPicker) childView;
+		}
+		return null;
+	}
+
+	private void recreateAllColumns()
+	{
+		// Fetch the picker proxy.
+		PickerProxy pickerProxy = getPickerProxy();
+		if (pickerProxy == null) {
+			return;
+		}
+
+		// Fetch the view group hosting all picker column child views.
+		TiCompositeLayout layout = (TiCompositeLayout) getNativeView();
+		if (layout == null) {
+			return;
+		}
+
+		// Remove all previously created column pickers.
+		layout.removeAllViews();
+
+		// Create new column picker views and add them to layout.
+		PickerColumnProxy[] columnProxies = pickerProxy.getColumns();
+		for (int columnIndex = 0; columnIndex < columnProxies.length; columnIndex++) {
+			// Create the column picker.
+			NumberPicker pickerView = new NumberPicker(pickerProxy.getActivity());
+			pickerView.setOnValueChangedListener(this);
+			pickerView.setWrapSelectorWheel(false);
+			pickerView.setDescendantFocusability(NumberPicker.FOCUS_BLOCK_DESCENDANTS); // Disable text editor.
+			update(pickerView, columnProxies[columnIndex]);
+
+			// Select a row based on proxy's last assigned index.
+			int rowIndex = Math.max(pickerProxy.getSelectedRowIndex(columnIndex), 0);
+			pickerView.setValue(Math.min(rowIndex, pickerView.getMaxValue()));
+
+			// Add picker to layout.
+			layout.addView(pickerView);
+		}
+	}
+
+	private void update(NumberPicker pickerView, PickerColumnProxy columnProxy)
+	{
+		// Validate arguments.
+		if ((pickerView == null) || (columnProxy == null)) {
+			return;
+		}
+
+		// Fetch the picker proxy.
+		PickerProxy pickerProxy = getPickerProxy();
+		if (pickerProxy == null) {
+			return;
+		}
+
+		// Fetch the currently selected row index. To be restored after updating column.
+		int lastSelectedRowIndex = pickerView.getValue();
+
+		// Set all row titles in the picker.
+		PickerRowProxy[] rowProxyArray = columnProxy.getRows();
+		if (rowProxyArray.length > 0) {
+			pickerView.setDisplayedValues(TiConvert.toStringArray(rowProxyArray));
+			pickerView.setMaxValue(rowProxyArray.length - 1);
+		} else {
+			pickerView.setDisplayedValues(new String[] { " " });
```

Comment:
Is this space here so `String.isEmprty()` returns `false`, or could this be an empty string?

Reply:
The Java `NumberPicker` class renders empty strings wrong. It'll render the top row's text if the selected row's string is empty. Setting it to a single space works-around the issue.

I added a comment above noting why it's like this.